external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1416,20 +1422,51 @@ impl DependencyGraph {
             }
         }
 
-        let sub_graph = DependencyGraph::read_from_lock(
-            package_path.to_path_buf(),
-            from_id,
-            from_name,
-            &mut output.stdout.as_slice(),
-            Some(resolver),
-        )
-        .with_context(|| {
-            format!(
-                "Parsing response from '{resolver}' for dependency '{to_name}' of package '{from_id}'"
-            )
-        })?;
+        let mut result = Vec::new();
+        let mut reader = BufReader::new(output.stdout.as_slice());
+        let mut buffer = Vec::new();
+        // Loop over null-separated lock file contents, creating the graph and adding it to the result.
+        loop {
+            match reader.read_until(0, &mut buffer) {
+                Ok(0) => break, // EOF
+                Ok(_) => {
+                    // Remove the null byte if it's present
+                    if buffer.last() == Some(&0) {
+                        buffer.pop();
+                    }
+
+                    let sub_graph = DependencyGraph::read_from_lock(
+                        package_path.to_path_buf(),
+                        from_id,
+                        from_name,
+                        &mut buffer.as_slice(),
+                        Some(resolver),
+                    ).with_context(|| {
+                        format!("Parsing response from '{resolver}' for dependency '{to_name}' of package '{from_id}'")
+                    })?;
+
+                    let mut root_sub_package = None;
+                    for (_from_package, to_pkg, _) in sub_graph.package_graph.edges(from_id) {
+                        if root_sub_package.is_some() {
+                            bail!("Multiple root packages in external dependencies found but expected only one");
+                        }
```

Comment:
It's not clear to me why we need this restriction, vs introducing all the edges from the root of the sub-graph to the main graph. I think it's fine for now, because the only customer is MVR and we know it won't produce a manifest with multiple edges, but can you add a TODO here to relax that constraint that we can look at later?

Reply:
> not clear to me why we need this restriction [...] but can you add a TODO here to relax that constraint that we can look at later?

Good point, will do!